File: epi_judge_java/epi/TreeLevelOrder.java
package epi;
import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.List;

import java.util.LinkedList;
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;


public class TreeLevelOrder {
  @EpiTest(testDataFile = "tree_level_order.tsv")

  public static List<List<Integer>>
  binaryTreeDepthOrder(BinaryTreeNode<Integer> tree) {
	  
	  List<BinaryTreeNode<Integer>> stack = new ArrayList<>();
	  List<BinaryTreeNode<Integer>> stack2;
	  stack.add(tree);
	  
	  List<List<Integer>> answer = new ArrayList<>();
	  
	  while (!stack.isEmpty()) {
		  List<Integer> thisLevel = new ArrayList<>();
		  stack2 = new ArrayList<>();
		  for (var node: stack) {
			  if (node != null) {
				  thisLevel.add(node.data);
				  stack2.add(node.left);
				  stack2.add(node.right);
			  }
		  }
		  if (!thisLevel.isEmpty()) {
			  answer.add(thisLevel);
		  }
		  stack = stack2;
	  }
	  
	  return answer;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "TreeLevelOrder.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: epi_judge_python/tree_level_order.py
from typing import List

from binary_tree_node import BinaryTreeNode
from test_framework import generic_test

def bfs(ts):
  if not ts:
    return []
  values = [x.data for x in ts]
  children = [c for t in ts for c in (t.left, t.right) if c]
  return [values]+bfs(children)

def binary_tree_depth_order(tree: BinaryTreeNode) -> List[List[int]]:
  from collections import deque
  levels = []
  q = [tree] if tree else []
  while q:
    children = []
    level = []
    for t in q:
      level.append(t.data)
      children += [x for x in (t.left, t.right) if x]
    levels.append(level)
    q = children

  return levels

if __name__ == '__main__':
  exit(
    generic_test.generic_test_main('tree_level_order.py',
                                   'tree_level_order.tsv',
                                   binary_tree_depth_order))
